lightkurve/lightcurve.py

```diff
@@ -1703,6 +1708,120 @@ def errorbar(self, **kwargs):
+        waterfall_method : 'str'
+            Type of waterfall plot, choose from `'average'` or `'sigma'`.
+            If `'average'`, the plot will display the average value in each bin.
+            If `'sigma'`, the plot will display the average in the bin divided by
+            the error in each bin, in order to show the data in terms of standard
+            deviation.
+        kwargs : dict
+            Dictionary of arguments to be passed to matplotlib
+        Returns
+        -------
+        ax : matplotlib.axes._subplots.AxesSubplot
+            The matplotlib axes object.
+        """
+        bin_method = validate_method(bin_method, supported_methods=['mean', 'median'])
+        if bin_method == 'mean':
+            bin_func = lambda y, e: (np.nanmean(y), np.sum(e**2)**0.5/len(e))
+        elif bin_method == 'median':
+            bin_func = lambda y, e: (np.nanmedian(y), np.sum(e**2)**0.5/len(e))
+
+        waterfall_method = validate_method(waterfall_method, supported_methods=['average', 'sigma'])
+        if waterfall_method == 'sigma':
+            waterfall_func = lambda y, e: (y - 1)/e
+        elif waterfall_method == 'average':
+            waterfall_func = lambda y, e: y
+        else:
+            raise ValueError('`waterfall_method` not understood.')
+
+        x, y, e = self.time_original[np.argsort(self.time_original)], self.flux[np.argsort(self.time_original)], self.flux_err[np.argsort(self.time_original)]
+        e /= np.nanmedian(self.flux)
+        y /= np.nanmedian(self.flux)
+
+        n = int(self.period/np.median(np.diff(x)) * (phmax - phmin)/bin_points)
+        ph = x/self.period % 1
+        cyc = np.asarray((x - x % self.period)/self.period, int)
+        cyc -= np.min(cyc)
+
+        phase = (self.t0 % self.period) / self.period
+        ph = ((x - (phase * self.period)) / self.period) % 1
+        cyc = np.asarray((x - ((x - phase * self.period) % self.period))/self.period, int)
+        cyc -= np.min(cyc)
+        ph[ph > 0.5] -= 1
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            ar = np.zeros((n, np.max(cyc) + 1)) * np.nan
+            bs = np.linspace(phmin, phmax, n)
+            cycs = np.arange(0, np.max(cyc) + 1)
+            for cyc1 in np.unique(cyc):
+                k = cyc == cyc1
+                for jdx in range(n-1):
+                    j = (ph[k] > bs[jdx]) & (ph[k] <= bs[jdx+1])
+                    if j.sum() == 0:
+                        ar[jdx, cyc1] = np.nan
+                    else:
+                        ar[jdx, cyc1] = waterfall_func(*bin_func(y[k][j], e[k][j]))
+                        if waterfall_method == 'average':
+                            # If the method is average we need to denormalize the plot
+                            ar[jdx, cyc1] *= np.nanmedian(self.flux)
+
+
+        d = np.max([np.abs(np.nanmedian(ar) - np.nanpercentile(ar, 5)), np.abs(np.nanmedian(ar) - np.nanpercentile(ar, 95))])
```

Comment:
:point_up: Two consecutive line breaks are not allowed inside methods/functions.